frontend/packages/git-service/src/services/hooks.ts

```diff
@@ -0,0 +1,79 @@
+import * as React from 'react';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+import { SecretKind } from '@console/internal/module/k8s';
+import { GitSource, SecretType as GitSecretType, GitProvider } from '../types';
+import { detectDeployStrategies, DetectedStrategy } from '../utils/deploy-strategy-detector';
+import { detectGitProvider } from '../utils/git-provider-detector';
+import { getGitService } from './git-service';
+
+type DetectedServiceData = {
+  loaded: boolean;
+  loadError?: any;
+  gitProvider: GitProvider;
+  strategies: DetectedStrategy[];
+};
+
+const gitServiceFromSource = (
+  gitProvider: GitProvider,
+  repository: string,
+  ref?: string,
+  contextDir?: string,
+  secret?: SecretKind,
+) => {
+  let secretType: GitSecretType;
+  let secretContent: any;
+  switch (secret?.type) {
+    case SecretType.basicAuth:
+      secretType = GitSecretType.BASIC_AUTH;
+      secretContent = secret.data;
+      break;
+    case SecretType.sshAuth:
+      secretType = GitSecretType.SSH;
+      secretContent = secret['ssh-privatekey'];
+      break;
+    default:
+      secretType = GitSecretType.NO_AUTH;
+  }
+  const gitSource: GitSource = {
+    url: repository,
+    ref,
+    contextDir,
+    secretType,
+    secretContent,
+  };
+  return getGitService(gitSource, gitProvider);
+};
+
+export const useDeployStrategyDetection = (
```

Comment:
Since we can have gitops and pipelines as code in future as strategy which are not deployment strategy technically.
```suggestion
export const useImportStrategyDetection = (
```

Reply:
Updated.